### PR TITLE
Docs: Automated tests in monorepo

### PR DESCRIPTION
**Pull request by brbrr**
Adds a documentation for automated tests in monorepo

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

n/a
#### Does this pull request change what data or activity we track or use?

n/a
#### Testing instructions:

* Read the docs.
* Does it helps?
* Does it lack something?

### CODE REVIEW

File: tools/e2e-commons/README.md [hunk 1]
```diff
@@ -1,20 +1,26 @@
...
+## Overview
 
 The scope of this library is to:
 
 - provide a consistent way to launch and configure a Jetpack environment for e2e testing
-- create test results, send Slack notifications
-- provide the most common pages modeled as page objects (see [Page objects model](https://playwright.dev/docs/test-pom)).
+- provide the most common pages modeled as page objects (see [Page objects model](https://playwright.dev/docs/test-pom))
 - provide implementation of the most common flows (login, connect Jetpack)
+- create test results, send Slack notifications
```
Comment (adimoldovan) [inline tools/e2e-commons/README.md]: Sending of Slack notifications is not available anymore in this library.

```suggestion
- create test results
```

File: docs/automated-testing.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+Monorepo provides support for `jest` as testing framework, and `@testing-library/react` as a testing library for React components.
+
+### React components
+
+There are examples scattered through the monorepo such as [connection-status-card](/projects/js-packages/connection/components/connection-status-card/test/component.jsx) card. Refer to [documentation](https://testing-library.com/docs/react-testing-library/intro) for more details.
+
+### Gutenberg blocks
+
+The [official core documentation](https://developer.wordpress.org/block-editor/contributors/code/testing-overview/) covers quite a lot of basics and would a good starting point for anyone starting with Gutenberg tests. There are multiple types of tests one can write for a blocks:
```
Comment (adimoldovan) [inline docs/automated-testing.md]:
```suggestion
The [official core documentation](https://developer.wordpress.org/block-editor/contributors/code/testing-overview/) covers quite a lot of basics and would a good starting point for anyone starting with Gutenberg tests. There are multiple types of tests one can write for a block:
```

File: docs/automated-testing.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+- [`validate` tests](/projects/plugins/jetpack/extensions/shared/test/block-fixtures.md) - Tests that verify expected block output.
+- `edit` tests - Tests for edit handlers.
+- `controls` tests - Tests for block controls.
+
+Subscriptions block has a good example of how to write tests for all of these types of tests. Refer to [subscriptions block tests](/projects/plugins/jetpack/extensions/blocks/subscriptions/test) for more details.
```
Comment (adimoldovan) [inline docs/automated-testing.md]:
```suggestion
Subscriptions block has a good example of how to write these types of tests. Refer to [subscriptions block tests](/projects/plugins/jetpack/extensions/blocks/subscriptions/test) for more details.
```

**Review by adimoldovan — COMMENTED** [review]
Left some small suggestions.

**Review by dereksmart — APPROVED** [review]

**Pull request merged**